src/resources/bootstrap.cpp

```diff
@@ -459,15 +448,31 @@ static bool do_bootstrap(buddy::resources::ProgressHook progress_hook) {
         return false;
     }
 
+    // clear installed resources
+    Path target_path("/internal/res");
+    buddy::resources::InstalledRevision::clear();
+    remove_directory_recursive(target_path);
+
     // copy the resources
     reporter.report("Copying files");
     source_path.set("/bbf");
-    Path target_path("/internal/res");
     if (!copy_resources_directory(source_path, target_path, reporter)) {
         log(LOG_SEVERITY_ERROR, "Failed to copy resources");
         return false;
     }
 
+    // calculate the hash of the resources we just installed
+    buddy::resources::Hash current_hash;
+    if (!buddy::resources::calculate_directory_hash(current_hash, "/internal/res")) {
+        log(LOG_SEVERITY_ERROR, "Failed to calculate hash of /internal/res directory");
+        return false;
+    }
+
+    if (buddy::resources::revision::required_revision.hash != current_hash) {
+        log(LOG_SEVERITY_ERROR, "Installed resources but the hash does not match!");
+        return false;
```

Comment:
I assume dealing with the invalid hash will be a subject of following PR's

Reply:
Yes! As you correctly pointed out in the comment bellow, we should handle some cases with broken filesystem etc. I plan to add this later.